content/en/blog/_posts/2024-12-11-windows-memory-pressure-eviction.md

```diff
@@ -0,0 +1,95 @@
+{{< /note >}}
+
+This approach differs from Linux, where available memory is reported based on _Working Set_ memory usage.
+
+## Why use commit memory on Windows?
+
+Before diving into the reasons for using commit memory on Windows for eviction signalling, it's helpful to understand the difference between _Working Set_ and _Commit_ memory.
+
+_Working Set_ memory represents the amount of physical memory currently used by processes. It includes the pages of memory that are actively in use and excludes memory pages that have been swapped out to disk (if paging is enabled).
+
+_Commit_ memory, on the other hand, represents the total amount of virtual memory that has been reserved by processes - including memory that has been swapped out to disk. When memory is "committed", it means that the system has allocated the backing storage (either physical RAM or space in the paging file(s)) to ensure that the memory will be available if needed.
+
+There are two reasons to using the global commit levels for measuring the available memory
+on a Windows node:
+
+1. **Memory limits for containers in Pods running on Windows nodes are enforced using Commit (not Working Set) memory**.
+   Windows memory management enforces limits on processes based on the total amount of memory committed,
+   not just the memory in use. 
+   This also means that memory limits specified for containers in Kubernetes Pods are enforced
+   on the total amount of virtual memory committed by all processes in the container. 
+   Consistency in how memory usage is tracked and enforced is essential for workload stability.
+
+2. **Paging (swap) is automatically enabled on Windows**, and it plays an essential role in the operating system's memory management. Disabling paging is not recommended because it can lead to system instability.
+
+## Eviction thresholds
+
+By default, Windows nodes have a "hard" eviction threshold of 500Mi
+(kubelet setting `--eviction-hard=memory.available<500Mi`).
+This means that if the system's availble commit memory is less than 500 Mi,
+the kubelet will try to evict pods.
+
+For more information on configuring memory-pressure based eviction and understanding the eviction signals and thresholds,
+refer to the official Kubernetes documentation on [memory-pressure-eviction](/docs/concepts/scheduling-eviction/node-pressure-eviction/#eviction-signals-and-thresholds).
+
+
+## Monitoring commit memory for Windows nodes
+
+Understanding the distinction between _Working Set_ and _Commit_ memory and being able to monitor both is critical for maintaining the stability of individual Pod workloads and the entire Windows node.
+
+Currently, `kubectl top` continues to report Working Set memory. As a result, a Pod may be evicted for exceeding its memory requests or encounter memory allocation failures due to exceeding its `CommitLimit`, even when the _Working Set_ memory usage appears to be well within the expected bounds. This discrepancy can lead to confusion.
```

Comment:
Did you mean:
```suggestion
Before v1.31, the values you saw from `kubectl top` reported Working Set memory. As a result, a Pod could have been evicted for exceeding its memory requests, or a Pod might have encountered memory allocation failures (due to exceeding its `CommitLimit`) - and this could happen even when the _Working Set_ memory usage appeared to be well within the expected bounds. This discrepancy could have led to confusion.
```

It's not clear if v1.31 improved things.